### PR TITLE
Implement `dataset_config.py` to inject dynamic model responses into static-networks

### PR DESCRIPTION
**Pull request by ZhuokunDing**

### CODE REVIEW

File: neuro_data/static_images/data_schemas.py [hunk 1]
```diff
@@ -938,6 +945,11 @@ def fill():
                 StaticMultiDataset.Member().insert1(dict(scan, name = _template.format(**scan)), ignore_extra_fields=True)
 
     def fetch_data(self, key, key_order=None):
+        from neuro_data.static_images.dataset_config import MultiDataset
+        key = (self & key).fetch1()
+        if key in MultiDataset:
```
Comment (Zhiweid) [inline neuro_data/static_images/data_schemas.py:950]: Is it possible to have a valid dataset both in MultiDataset as well as in InputResponse? If the only case where we use MultiDataset is when data source comes from in silico then we may want to assert scan_type == 'in_silico' in StaticScan if fetching data from MultiDataset

File: neuro_data/static_images/requests.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+from os import sched_getaffinity
```
Comment (Zhiweid) [inline neuro_data/static_images/requests.py:1]: What is this requests.py module for?

File: neuro_data/static_images/dataset_config.py [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+
+    class NoBehFullFieldFrame(dj.Part):
+        # implemented only for full field stimulus with stimulus_type=='stimulus.Frame'
+        definition = """
+        # mimic the behavior of InputResponse with the corresponding preprocess_id to compute the statistics of input and responses, do not compute stats for behavior variables
+        -> master
+        ---
+        stats_tier="train"                 : enum("train", "test", "validation", "all")                # tier used for computing stats
+        stats_per_input                    : tinyint                                                  # whether to compute stats per input
+        linear_monitor                     : tinyint                                                  # whether to compute linear monitor
```
Comment (Zhiweid) [inline neuro_data/static_images/dataset_config.py:231]: I think you are referring to the 'gamma' secondary key in preprocessing, not 'linear_monitor'. 'gamma' is the indicates whether to convert inputs to the luminance space. Linearized monitor is a different thing. We can include both.

File: neuro_data/static_images/dataset_config.py [hunk 4]
```diff
@@ -0,0 +1,533 @@
...
+            # reshape inputs
+            images = np.stack(images)
+            if len(images.shape) == 3:
+                log.info("Adding channel dimension")
+                images = images[:, None, ...]
+            elif len(images.shape) == 4:
+                images = images.transpose(0, 3, 1, 2)
+
+            # gamma correction
+            if key["linear_monitor"]:
```
Comment (Zhiweid) [inline neuro_data/static_images/dataset_config.py:299]: this should be key['gamma']. And we probably want to pend this part until we agree on a luminance conversion table that everyone uses. This table multi_mei.ClosestCalibration is very outdated.

File: neuro_data/static_images/dataset_config.py [hunk 5]
```diff
@@ -0,0 +1,533 @@
...
+                trial_idx=trial_idx.astype(np.uint32),
+                neurons=neurons,
+                tiers=tiers.astype("S"),
+                statistics=statistics,
+            )
+
+
+@schema
+class DatasetInputResponse(dj.Computed):
+    # Inject datasets back to InputResponse table. Mutltiple datasets with the same
```
Comment (Zhiweid) [inline neuro_data/static_images/dataset_config.py:456]: may want to note that only insert back to the master table of InputResponse and the part tables would be empty

File: neuro_data/static_images/requests.py (hunk 2 above)
Comment (ZhuokunDing) [reply neuro_data/static_images/requests.py:1]: This is for providing the `content` attribute for the `DataConfig` part tables without putting all data in the source code.

File: neuro_data/static_images/dataset_config.py (hunk 5 above)
Comment (ZhuokunDing) [reply neuro_data/static_images/dataset_config.py:456]: good point, will do

File: neuro_data/static_images/dataset_config.py (hunk 3 above)
Comment (ZhuokunDing) [reply neuro_data/static_images/dataset_config.py:231]: After a second thought, I think it is better to include those parameters in InputConfig instead.

**Review by jiakunf — APPROVED** [review]

**Review by Zhiweid — APPROVED** [review]

**Pull request merged**